base/applications/cmdutils/timeout/timeout.c

```diff
@@ -164,7 +164,11 @@ INT InputWait(BOOL bNoBreak, INT timerValue)
                      * We therefore use the __emul() intrinsic which does
                      * the correct job.
                      */
+#if !defined(_M_ARM)
                     DueTime.QuadPart = __emul(1000 - timeElapsed, -10000);
```

Comment:
You can remove this, since Int32x32To64 is defined to that on the appropriate architectures already.

Reply:
According to the comment above it, this did not work as expected.

Reply:
It may work on latest msvc versions but not "all".

Reply:
Our headers have
```
/* 32 to 64 bit multiplication. GCC is really bad at optimizing the native math */
#if defined(_M_IX86) && defined(__GNUC__) && \
    !defined(MIDL_PASS)&& !defined(RC_INVOKED) && !defined(_M_CEE_PURE)
 #define Int32x32To64(a,b) __emul(a,b)
 #define UInt32x32To64(a,b) __emulu(a,b)
#else
 #define Int32x32To64(a,b) (((__int64)(long)(a))*((__int64)(long)(b)))
 #define UInt32x32To64(a,b) ((unsigned __int64)(unsigned int)(a)*(unsigned __int64)(unsigned int)(b))
#endif
```
Removing the ```__GCC__``` should fix it.